lib/sles4sap/ipaddr2.pm

```diff
@@ -94,6 +94,34 @@ sub get_private_ip_range {
     return %range;
 }
 
+=head2 az_img_from_vhd_create
+
+    my $image = az_img_from_vhd_create(resource_group => $rg, name => $name, source => $uploaded_vhd_url);
+
+Create an image out of a .vhd disk in Azure storage.
+
+=over
+
+=item B<resource_group> - existing resource group
+
+=item B<name> - NAME of the created image
+
+=item B<source> - URI of the vhd disk from which the image will be created
+
+=back
+=cut
+
+sub az_img_from_vhd_create {
+    my (%args) = @_;
+    my $az_cmd = join(' ', 'az image create',
+        '--resource-group', $args{resource_group},
+        '-n', $args{name},
+        '--os-type', 'linux',
+        '--source', $args{source});
+    assert_script_run($az_cmd, timeout => 600);
+    return $args{name};
```

Comment:
This function is returning a value it receives without modifying it, calculating it, or anything similar; more than a function, it's a procedure. I think this can be removed from the function, and below after the procedure is called, simply do `$args{os} = $rg . 'img';` or something similar:
```
if ($args{os} =~ /\.vhs$/) {
    my $img_name = $rg . 'img';
    az_img_from_vhd_create(resource_group => $rg, $name => $img_name, source => $args{os});
    # if az_img_from_vhd_create worked, image has been created from the vhd blob.
    # update the os arg with it to use image instead
    $args{os} = $img_name;
}
```

Reply:
Done